tests/common/test_gtsfm_data.py

```diff
@@ -0,0 +1,149 @@
+
+        cam_idx = 0
+        test_data = GtsfmData(1)
+        test_data.add_camera(expected, cam_idx)
+
+        computed = test_data.get_camera(cam_idx)
+        self.assertTrue(computed.equals(expected, EQUALITY_TOLERANCE))
+
+    def test_get_camera_invalid(self):
+        """Test for get_camera for an index where the camera does not exist."""
+
+        computed = EXAMPLE_DATA.get_camera(10)
+        self.assertIsNone(computed)
+
+    def test_get_track(self):
+        """Testing getter for track."""
+        expected_track = SfmTrack(np.array([6.41689062, 0.38897032, -23.58628273]))
+        expected_track.add_measurement(0, np.array([383.88000488, 15.2999897]))
+        expected_track.add_measurement(1, np.array([559.75, 106.15000153]))
+
+        computed = EXAMPLE_DATA.get_track(1)
+
+        # comparing just the point because track equality is failing
+        np.testing.assert_allclose(computed.point3(), expected_track.point3())
+
+    def test_add_track_with_valid_cameras(self):
+        """Testing track addition when all cameras in track are already present."""
+
+        gtsfm_data = copy.deepcopy(EXAMPLE_DATA)
+
+        # add a track on camera #0 and #1, which exists in the data
+        track_to_add = SfmTrack(np.array([0, -2.0, 5.0]))
+        track_to_add.add_measurement(0, np.array([20.0, 5.0]))
+        track_to_add.add_measurement(1, np.array([60.0, 50.0]))
+
+        self.assertTrue(gtsfm_data.add_track(track_to_add))
+
+    def test_add_track_with_nonexistant_cameras(self):
+        """Testing track addition where some cameras are not in tracks, resulting in failure."""
+        gtsfm_data = copy.deepcopy(EXAMPLE_DATA)
+
+        # add a track on camera #0 and #1, which exists in the data
+        track_to_add = SfmTrack(np.array([0, -2.0, 5.0]))
+        track_to_add.add_measurement(0, np.array([20.0, 5.0]))
+        track_to_add.add_measurement(3, np.array([60.0, 50.0]))  # this camera does not exist
+
+        self.assertFalse(gtsfm_data.add_track(track_to_add))
+
+    def test_drop_cameras(self):
+        """Test dropping cameras."""
+
+        # drop the camera at index 1 in the example data
+        computed = EXAMPLE_DATA.drop_cameras([1])
+
+        self.assertEqual(computed.get_camera(0), computed.get_camera(0))
+        self.assertEqual(computed.get_camera(2), computed.get_camera(2))
+        self.assertEqual(computed.number_tracks(), 0)
+
+    @mock.patch.object(graph_utils, "get_nodes_in_largest_connected_component", return_value=[1, 2, 4])
+    def test_select_largest_connected_component(self, graph_largest_cc_mock):
+        """Test pruning to largest connected component according to tracks."""
+        gtsfm_data = GtsfmData(5)
+        cam = PinholeCameraCal3Bundler(Pose3(), Cal3Bundler())
+
+        # add the same camera at all indices
+        for cam_idx in range(gtsfm_data.number_images()):
+            gtsfm_data.add_camera(cam, cam_idx)
+
+        # add two tracks to define connected component
```

Comment:
inline comment might be nice, explaining that this track will be thrown away (and we should check that it is), and that these cameras will be dropped

Reply:
better?

Reply:
yes, thanks!